### PR TITLE
[stdlib] RawRepresentable: revert to default _rawHashValue(seed:)

### PR DESCRIPTION
**Pull request by lorentey**
https://bugs.swift.org/browse/SR-10734
rdar://problem/51319164

### CODE REVIEW

**Comment by lorentey** [comment]
@swift-ci smoke test

**Review by jrose-apple — APPROVED** [review]
Sorry for the trouble…

File: test/stdlib/RawRepresentable-tricky-hashing.swift [hunk 1]
```diff
@@ -30,27 +27,56 @@ struct TrickyRawRepresentable: RawRepresentable, Hashable {
...
+
+  func hash(into hasher: inout Hasher) {
+    hasher.combine(23)
   }
 }
 
-// CHECK: hash(into:) matches: true
-print("hash(into:) matches: \(Hasher.hash(s1) == Hasher.hash(s2))")
+suite.test("Custom hashing") {
+  // In 5.0, RawRepresentable had with a bogus default implementation of
```
Comment (lorentey) [inline test/stdlib/RawRepresentable-tricky-hashing.swift]: typo

File: test/stdlib/RawRepresentable-tricky-hashing.swift [hunk 2]
```diff
@@ -30,27 +27,56 @@ struct TrickyRawRepresentable: RawRepresentable, Hashable {
...
-let s1 = TrickyRawRepresentable(rawValue: "café")!
-let s2 = TrickyRawRepresentable(rawValue: "cafe\u{301}")!
+suite.test("Tricky hashing") {
+  // RawRepresentable is not Equatable itself, but it does provide a generic
+  // implementation of == based on rawValues. This gets picked up as the
+  // implementation of Equatable.== when a concrete RawRepresentable type
+  // conforms to Equatable without providing its own implementation.
+  //
+  // However, RawRepresentable used to not provide equivalent implementations
+  // for hashing, allowing the compiler to synthesized hashing as usual, based
```
Comment (lorentey) [inline test/stdlib/RawRepresentable-tricky-hashing.swift]: typo

**Comment by lorentey** [comment]
@jrose-apple It was me who introduced this bug -- I tried to be overly clever with single shot hashing. :/

**Comment by jrose-apple** [comment]
Huh, my mistake must have been around something else then.

…don't we *still* have it wrong if they only override `hash(into:)` and not `hashValue` or vice versa? ><

**Comment by lorentey** [comment]
Yeah, RawRepresentable types that only implement `hashValue` remain broken.

(And they won't even get the usual warning about only implementing hashValue -- we probably should warn if ==/hash(into:)/hashValue do not all come from the same level. Or just warn on hashValue implementations in general?)

**Comment by lorentey** [comment]
The original sin was adding the generic `==` definition for RawRepresentable, and that goes back to before the dawn of time itself.

**Comment by lorentey** [comment]
```
17:15:37 FAIL: Swift(linux-x86_64) :: stdlib/RawRepresentable-tricky-hashing.swift (7381 of 12050)
17:15:37 ******************** TEST 'Swift(linux-x86_64) :: stdlib/RawRepresentable-tricky-hashing.swift' FAILED ********************
[...]
17:15:37 error: no check strings found with prefix 'CHECK:'
```

🤦‍♂️

**Comment by lorentey** [comment]
@swift-ci smoke test

**Comment by lorentey** [comment]
```
12:22:24 FAIL: Swift(macosx-x86_64) :: PrintAsObjC/extensions.swift (9855 of 12114)
12:22:24 ******************** TEST 'Swift(macosx-x86_64) :: PrintAsObjC/extensions.swift' FAILED ********************
12:22:24 Script:
12:22:24 --
12:22:24 : 'RUN: at line 3';   rm -rf "/Users/buildnode/jenkins/workspace/swift-PR-osx-smoke-test/branch-master/buildbot_incremental/swift-macosx-x86_64/test-macosx-x86_64/PrintAsObjC/Output/extensions.swift.tmp" && mkdir -p "/Users/buildnode/jenkins/workspace/swift-PR-osx-smoke-test/branch-master/buildbot_incremental/swift-macosx-x86_64/test-macosx-x86_64/PrintAsObjC/Output/extensions.swift.tmp"
12:22:24 : 'RUN: at line 4';   /Users/buildnode/jenkins/workspace/swift-PR-osx-smoke-test/branch-master/buildbot_incremental/swift-macosx-x86_64/bin/swiftc -frontend -target x86_64-apple-macosx10.9  -module-cache-path '/Users/buildnode/jenkins/workspace/swift-PR-osx-smoke-test/branch-master/buildbot_incremental/swift-macosx-x86_64/swift-test-results/x86_64-apple-macosx10.9/clang-module-cache' -sdk '/Applications/Xcode-beta.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk' -swift-version 4  -typo-correction-limit 10   -enable-source-import -sdk '/Users/buildnode/jenkins/workspace/swift-PR-osx-smoke-test/branch-master/swift/test/Inputs/clang-importer-sdk' -I '/Users/buildnode/jenkins/workspace/swift-PR-osx-smoke-test/branch-master/swift/test/Inputs/clang-importer-sdk/swift-modules'   -disable-sil-ownership-verifier -emit-module -o /Users/buildnode/jenkins/workspace/swift-PR-osx-smoke-test/branch-master/buildbot_incremental/swift-macosx-x86_64/test-macosx-x86_64/PrintAsObjC/Output/extensions.swift.tmp /Users/buildnode/jenkins/workspace/swift-PR-osx-smoke-test/branch-master/swift/test/PrintAsObjC/extensions.swift -disable-objc-attr-requires-foundation-module
12:22:24 : 'RUN: at line 5';   /Users/buildnode/jenkins/workspace/swift-PR-osx-smoke-test/branch-master/buildbot_incremental/swift-macosx-x86_64/bin/swiftc -frontend -target x86_64-apple-macosx10.9  -module-cache-path '/Users/buildnode/jenkins/workspace/swift-PR-osx-smoke-test/branch-master/buildbot_incremental/swift-macosx-x86_64/swift-test-results/x86_64-apple-macosx10.9/clang-module-cache' -sdk '/Applications/Xcode-beta.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk' -swift-version 4  -typo-correction-limit 10   -enable-source-import -sdk '/Users/buildnode/jenkins/workspace/swift-PR-osx-smoke-test/branch-master/swift/test/Inputs/clang-importer-sdk' -I '/Users/buildnode/jenkins/workspace/swift-PR-osx-smoke-test/branch-master/swift/test/Inputs/clang-importer-sdk/swift-modules'   -disable-sil-ownership-verifier -parse-as-library /Users/buildnode/jenkins/workspace/swift-PR-osx-smoke-test/branch-master/buildbot_incremental/swift-macosx-x86_64/test-macosx-x86_64/PrintAsObjC/Output/extensions.swift.tmp/extensions.swiftmodule -typecheck -emit-objc-header-path /Users/buildnode/jenkins/workspace/swift-PR-osx-smoke-test/branch-master/buildbot_incremental/swift-macosx-x86_64/test-macosx-x86_64/PrintAsObjC/Output/extensions.swift.tmp/extensions.h -import-objc-header /Users/buildnode/jenkins/workspace/swift-PR-osx-smoke-test/branch-master/swift/test/PrintAsObjC/../Inputs/empty.h -disable-objc-attr-requires-foundation-module
12:22:24 : 'RUN: at line 6';   '/usr/bin/python' '/Users/buildnode/jenkins/workspace/swift-PR-osx-smoke-test/branch-master/swift/utils/PathSanitizingFileCheck' --sanitize BUILD_DIR='/Users/buildnode/jenkins/workspace/swift-PR-osx-smoke-test/branch-master/buildbot_incremental/swift-macosx-x86_64' --sanitize SOURCE_DIR='/Users/buildnode/jenkins/workspace/swift-PR-osx-smoke-test/branch-master/swift' --use-filecheck '/Users/buildnode/jenkins/workspace/swift-PR-osx-smoke-test/branch-master/buildbot_incremental/llvm-macosx-x86_64/bin/FileCheck'  /Users/buildnode/jenkins/workspace/swift-PR-osx-smoke-test/branch-master/swift/test/PrintAsObjC/extensions.swift < /Users/buildnode/jenkins/workspace/swift-PR-osx-smoke-test/branch-master/buildbot_incremental/swift-macosx-x86_64/test-macosx-x86_64/PrintAsObjC/Output/extensions.swift.tmp/extensions.h
12:22:24 : 'RUN: at line 7';   '/usr/bin/python' '/Users/buildnode/jenkins/workspace/swift-PR-osx-smoke-test/branch-master/swift/utils/PathSanitizingFileCheck' --sanitize BUILD_DIR='/Users/buildnode/jenkins/workspace/swift-PR-osx-smoke-test/branch-master/buildbot_incremental/swift-macosx-x86_64' --sanitize SOURCE_DIR='/Users/buildnode/jenkins/workspace/swift-PR-osx-smoke-test/branch-master/swift' --use-filecheck '/Users/buildnode/jenkins/workspace/swift-PR-osx-smoke-test/branch-master/buildbot_incremental/llvm-macosx-x86_64/bin/FileCheck'  --check-prefix=NEGATIVE /Users/buildnode/jenkins/workspace/swift-PR-osx-smoke-test/branch-master/swift/test/PrintAsObjC/extensions.swift < /Users/buildnode/jenkins/workspace/swift-PR-osx-smoke-test/branch-master/buildbot_incremental/swift-macosx-x86_64/test-macosx-x86_64/PrintAsObjC/Output/extensions.swift.tmp/extensions.h
12:22:24 : 'RUN: at line 8';   '/Users/buildnode/jenkins/workspace/swift-PR-osx-smoke-test/branch-master/buildbot_incremental/llvm-macosx-x86_64/bin/clang' -fmodules-cache-path='/Users/buildnode/jenkins/workspace/swift-PR-osx-smoke-test/branch-master/buildbot_incremental/swift-macosx-x86_64/swift-test-results/x86_64-apple-macosx10.9/clang-module-cache' -fsyntax-only -x objective-c-header -fobjc-arc -fmodules -fmodules-validate-system-headers -Weverything -Werror -Wno-unused-macros -Wno-incomplete-module -Wno-auto-import -F '/Users/buildnode/jenkins/workspace/swift-PR-osx-smoke-test/branch-master/swift/test/Inputs/clang-importer-sdk'/frameworks -I /Users/buildnode/jenkins/workspace/swift-PR-osx-smoke-test/branch-master/buildbot_incremental/swift-macosx-x86_64/include -isysroot '/Users/buildnode/jenkins/workspace/swift-PR-osx-smoke-test/branch-master/swift/test'/Inputs/clang-importer-sdk /Users/buildnode/jenkins/workspace/swift-PR-osx-smoke-test/branch-master/buildbot_incremental/swift-macosx-x86_64/test-macosx-x86_64/PrintAsObjC/Output/extensions.swift.tmp/extensions.h
12:22:24 --
12:22:24 Exit Code: 70
12:22:24 
12:22:24 Command Output (stderr):
12:22:24 --
12:22:24 fatal error: error in backend: Unexpected end of file
12:22:24 
clang-7: error: clang frontend command failed with exit code 70 (use -v to see invocation)
12:22:24 Apple clang version 7.0.0 (https://github.com/apple/swift-clang.git 54d8dfd1a03ffabb107be7ea0d475c43ce25bc32) (https://github.com/apple/swift-llvm.git 8d110eebee5f8218a3ca87da92f61a2d4e49c307) (based on LLVM 7.0.0)
12:22:24 Target: x86_64-apple-darwin18.6.0
12:22:24 Thread model: posix
12:22:24 InstalledDir: /Users/buildnode/jenkins/workspace/swift-PR-osx-smoke-test/branch-master/buildbot_incremental/llvm-macosx-x86_64/bin
12:22:24 clang-7: note: diagnostic msg: PLEASE submit a bug report to https://bugs.llvm.org/ and include the crash backtrace, preprocessed source, and associated run script.
12:22:24 clang-7: note: diagnostic msg: 
12:22:24 ********************
12:22:24 
12:22:24 PLEASE ATTACH THE FOLLOWING FILES TO THE BUG REPORT:
12:22:24 Preprocessed source(s) and associated run script(s) are located at:
12:22:24 clang-7: note: diagnostic msg: /var/folders/_8/79jmzf2142z2xydc_01btlx00000gn/T/lit_tmp_DywuxP/extensions-ac9bb0.h
12:22:24 clang-7: note: diagnostic msg: /var/folders/_8/79jmzf2142z2xydc_01btlx00000gn/T/lit_tmp_DywuxP/extensions-ac9bb0.cache
12:22:24 clang-7: note: diagnostic msg: /var/folders/_8/79jmzf2142z2xydc_01btlx00000gn/T/lit_tmp_DywuxP/extensions-ac9bb0.sh
12:22:24 clang-7: note: diagnostic msg: Crash backtrace is located in
12:22:24 clang-7: note: diagnostic msg: /Users/buildnode/Library/Logs/DiagnosticReports/clang-7_<YYYY-MM-DD-HHMMSS>_<hostname>.crash
12:22:24 clang-7: note: diagnostic msg: (choose the .crash file that corresponds to your crash)
12:22:24 clang-7: note: diagnostic msg: 
12:22:24 
12:22:24 ********************
12:22:24 
12:22:24 --
12:22:24 
12:22:24 ********************
```

**Comment by lorentey** [comment]
@swift-ci smoke test macOS platform

**Comment by jrose-apple** [comment]
We still haven't figured that one out yet. :-(

**Pull request merged**